crates/sui-faucet/src/faucet/simple_faucet.rs

```diff
@@ -232,46 +254,37 @@ impl SimpleFaucet {
         info!(?uuid, ?coin_id, "Recycled coin");
     }
 
-    async fn execute_pay_sui_txn_with_retrials(
+    async fn execute_pay_sui_txn_with_retries(
         &self,
         coin_id: ObjectID,
         signer: SuiAddress,
         recipient: SuiAddress,
         amounts: &[u64],
         budget: u64,
         uuid: Uuid,
-    ) -> Result<SuiTransactionResponse, anyhow::Error> {
-        let retry_intervals_ms = [Duration::from_millis(500), Duration::from_millis(1000)];
-        let mut retry_iter = retry_intervals_ms.iter();
-        let mut res = self
-            .execute_pay_sui_txn(coin_id, signer, recipient, amounts, budget, uuid)
-            .await;
-        while res.is_err() {
-            if let Some(interval) = retry_iter.next() {
-                tokio::time::sleep(*interval).await;
-                info!(
-                    ?recipient,
-                    ?coin_id,
-                    ?uuid,
-                    "Retrying executing PaySui transaction in faucet, previous error: {:?}",
-                    &res,
-                );
-                res = self
-                    .execute_pay_sui_txn(coin_id, signer, recipient, amounts, budget, uuid)
-                    .await;
-            } else {
-                warn!(
-                    ?recipient,
-                    ?coin_id,
-                    ?uuid,
-                    "Failed to execute PaySui transactions in faucet with {} retries and intervals {:?}",
-                    retry_intervals_ms.len(),
-                    &retry_intervals_ms
-                );
-                break;
+    ) -> SuiTransactionResponse {
+        let mut retry_delay = Duration::from_millis(500);
+        loop {
+            let res = self
+                .execute_pay_sui_txn(coin_id, signer, recipient, amounts, budget, uuid)
```

Comment:
This function creates a fresh transaction each time, and signs it, instead we need to create and sign the transaction once, and repeatedly send it to the quorum driver. I.e. pull everything in execute_pay_sui_txn until execute_transaction outside this loop.

Reply:
@amnn  is right that we'd usually like to keep using the exactly same transaction. In this case, the transaction is expected to the same even though we built it multiple times, because all the params are the same. The only possible variable is object version, which SHOULDN'T change in good days.

That said, it's still better to build the tx once and send the clone, assuming faucet is the only operator of this account (it should be)

Reply:
I did consider doing this but convinced myself it would be impossible for the tx to change from one construction to the next. But if everyone is in agreement then i'll do it that way.